### PR TITLE
save and load model in runtime library

### PR DESCRIPTION
**Pull request by Yancey0623**
c.f. #2667

### CODE REVIEW

File: python/runtime/model/db.py [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+
+def _drop_table_if_exists(conn, table):
+    sql = "DROP TABLE IF EXISTS {0}".format(table)
+    cursor = conn.cursor()
+    cursor.execute(sql)
+
+
+def write_with_generator(datasource, table, gen):
+    """Write data into a table, the written data
+    comes from the input generator.
```
Comment (shendiaomo) [inline python/runtime/model/db.py:47]: Please add explanations for the parameters.

File: python/runtime/model/db.py [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+            w.write(row)
+            idx += 1
+
+    conn.close()
+
+
+def read_with_generator(datasource, table):
+    """Read data from a table, this function returns
+    a generator to yield the data.
+    """
```
Comment (shendiaomo) [inline python/runtime/model/db.py:84]: ditto.

File: python/runtime/model/model.py [hunk 3]
```diff
@@ -53,43 +65,67 @@ def __init__(self, typ, meta):
...
+
+            return _gen
+
+        write_with_generator(datasource, table, _bytes_reader(tarball))
+
+
+def load(datasource, table, cwd="./"):
+    """Load saved model from DBMS and restructure the Model object.
+    Args:
+
```
Comment (shendiaomo) [inline python/runtime/model/model.py:99]: We don't need these blank lines in the docstring?

File: python/runtime/model/model.py [hunk 4]
```diff
@@ -15,7 +15,19 @@
...
-from runtime.diagnostics import SQLFlowDiagnostic
+from runtime.model.db import read_with_generator, write_with_generator
+from runtime.model.tar import unzip_dir, zip_dir
+
+try:
+    import cPickle as pickle
+except ModuleNotFoundError:
+    import pickle
+
+# achieve the current work director into a tarball
```
Comment (typhoonzero) [inline python/runtime/model/model.py]: archive

**Review by typhoonzero — DISMISSED** [review]
LGTM

**Review by typhoonzero — APPROVED** [review]

**Pull request merged**